main.nf

```diff
@@ -273,18 +293,21 @@ if (merge && metadata) {
 
   // Check missing metadata values
   // #############################
+  print(metadata_to_check)
+
   missing_values = metadata_to_check.findAll {
     it.values()[0] in inputs.collect{it['prefix']}
   }.inject([], { result, row ->
     factors = []
-    merge.each { factor ->
+    merge_assembly.each { factor ->
```

Comment:
Must do this check for quantification groups too.

Reply:
Oh sorry, I could indeed reproduce it on both the master and this branch.
On this branch however, the message shows, but merge group are not computed accordingly, and I get : 
```
Transcripts assembly will be done separately in each of the following groups:
  1: SRR6357070  SRR6357071  SRR6357073
  2: SRR6357072  SRR6357074  SRR6357075
  null: SRR6357076
```
So the merge value is null and not '' as expected by the test [here](https://github.com/FAANG/analysis-TAGADA/blob/1e77f3c195d2dfe0c23568d56a987286bdb5a1b3/main.nf#L333) but I don't see how it's linked to this PR at the moment.

Reply:
You should check the contents of `metadata_to_merge_for_assembly` with `metadata_to_merge_for_assembly.view()` to verify that your factors appear as intended.